### DISCUSSION TITLE
How does compile time compare to regular TeX compilation?

### DISCUSSION THREAD

**Discussion by VictorieeMan** [tectonic-typesetting/tectonic#1153] [Q&A] [answered]
I'm curious if anyone has bench marked this against something like MikText etc. Typically, building LaTeX documents is slow, which brings the natural question whether this is faster?

**Comment by kpym** [comment, marked as answer]
It is faster! In general `pdflatex` is faster than `xelatex` (probably because `xelatex` uses more compression by default). But `tectonic` is even faster than `pdflatex` (and it makes really smaller pdfs, just like `xelatex`). The slowest is `lualatex` (but it has its own advantages).
This was my first and main reason to switch to `tectonic`.

The problem is that `tectonic` gets slower and slower with each new version. I was thinking of publishing a benchmark and opening an issue about it, but I have no time for that right now :(

**Reply by pkgw** [reply]
For what it's worth, Tectonic's core TeX engine code really has changed very little over the past years, so in principle it should be running at around the same speed over time. But there are certainly lots of factors that could cause things to be slower. And this is why we do benchmarking! Note that Rust has built-in support for benchmarks with the `cargo bench` command and associated infrastructure.

**Reply by VictorieeMan** [reply]
Oh, wouldn't it be neat to add or link some benchmarking in the readme? Things like that are really good when researching what solutions to go with. Thanks for your update here btw!

**Comment by VictorieeMan** [comment]
Great to know! I think the idea of benchmarking could be a good thing for projects like this. The score could be kept and held in the repository, maybe even presented in the first page readme.


Thanks for your reply!

**Comment by CathedralCph** [comment]
To get insights into Rust I rewrote my nodejs microservice using xelatex into Rust using Tectonic runtime. I was so much looking forward to test it but nodejs 2x faster and for 10 parallel executions 8x faster, so not sure what to learn here
